files/common/etc/default/grub.d/override.cfg

```diff
@@ -98,3 +98,18 @@ GRUB_CMDLINE_LINUX_DEFAULT="$GRUB_CMDLINE_LINUX_DEFAULT elevator=noop"
 #
 GRUB_CMDLINE_LINUX_DEFAULT="$GRUB_CMDLINE_LINUX_DEFAULT crashkernel=256M,high"
 GRUB_CMDLINE_LINUX_DEFAULT="$GRUB_CMDLINE_LINUX_DEFAULT crashkernel=256M,low"
+
+#
+# Disable unnecessary zeroing of allocated pages.
+#
+# The 5.3 linux kernel adds a new feature which allows pages to be
+# zeroed when allocating or freeing them: init_on_alloc and
+# init_on_free. init_on_alloc is enabled by default on Ubuntu. ZFS
+# allocates and frees pages frequently (via the ABD structure), e.g. for
+# every disk access. The additional overhead of zeroing these pages is
+# significant. I measured a ~40% regression in performance of an
+# uncached "zfs send ... >/dev/null".
+#
+# Upstream commit: 6471384af2a6530696fc0203bafe4de41a23c9ef
+#
+GRUB_CMDLINE_LINUX_DEFAULT="$GRUB_CMDLINE_LINUX_DEFAULT init_on_alloc=0"
```

Comment:
Do we also need to disable `init_on_free`?

Reply:
init_on_free is disabled by default on Ubuntu.

Reply:
If we specifically need `init_on_free=0`, I think it might be good to set that explicitly here, just in case Ubuntu ever decides to change that default setting..